hoomd/md/methods/methods.py

```diff
@@ -327,29 +372,79 @@ class ConstantPressure(Thermostatted):
             the barostat.
             In Voigt notation,
             :math:`[S_{xx}, S_{yy}, S_{zz}, S_{yz}, S_{xz}, S_{xy}]`
-            :math:`[\mathrm{pressure}]`. Stress can be reset after the method
-            object is created. For example, an isotropic pressure can be set by
-            ``npt.S = 4.``
+            :math:`[\mathrm{pressure}]`.
+
+            .. rubric:: Examples:
+
+            .. code-block:: python
+
+                npt.S = 4.0
+
+            .. code-block:: python
+
+                npt.S = hoomd.variant.Ramp(A=1.0,
+                                           B=2.0,
+                                           t_start=0,
+                                           t_ramp=1_000_000)
 
         tauS (float): Coupling constant for the barostat
             :math:`[\mathrm{time}]`.
 
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                npt.tauS = 2.0
+
         couple (str): Couplings of diagonal elements of the stress tensor,
-            can be "none", "xy", "xz","yz", or "xyz".
+            can be 'none', 'xy', 'xz', 'yz', or 'xyz'.
+
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                npt.couple = 'none'
+
+        box_dof(list[bool]): Box degrees of freedom with six boolean elements in
+            the order [x, y, z, xy, xz, yz].
+
+            .. rubric:: Example:
+
+            .. code-block:: python
+
+                npt.box_dof = [False, False, True, False, False, False]
 
-        box_dof(list[bool]): Box degrees of freedom with six boolean elements
-            corresponding to x, y, z, xy, xz, yz, each.
+        rescale_all (bool): When True, rescale all particles, not only those
+            selected by the filter.
 
-        rescale_all (bool): if True, rescale all particles, not only those in
-            the group.
+            .. rubric:: Example:
 
-        gamma (float): Friction constant for the box degrees of freedom, Default
-            to 0  :math:`[\mathrm{time^{-1}}]`.
+            .. code-block:: python
+
+                npt.rescale_all = True
+
+        gamma (float): Friction constant for the box degrees of freedom
+            :math:`[\mathrm{time^{-1}}]`.
 
         barostat_dof (tuple[float, float, float, float, float, float]):
             Additional degrees of freedom for the barostat (:math:`\nu_{xx}`,
             :math:`\nu_{xy}`, :math:`\nu_{xz}`, :math:`\nu_{yy}`,
             :math:`\nu_{yz}`, :math:`\nu_{zz}`)
+
+            Save and restore the barostat degrees of freedom when continuing
+            simulations:
+
+            .. rubric:: Examples:
+
+            .. code-block:: python
+
+                saved_barostat_dof = npt.barostat_dof
+                # Save to a file.
```

Comment:
Would it be too much to write the save function here as well? Is this a problem for sybil
```suggestion
                # Save to a file.
                import numpy
                numpy.savetxt('barostat_dof.txt', saved_barostat_dof)
```

Reply:
This is not a problem with Sybil. I'm concerned about dictating a particular solution for something that users may be opinionated on and there are so many potential solutions (numpy, json, bson, signac job doc, signac data store, h5py). I'm also concerned about providing a solution that will "just work" for a high fraction of users that copy and paste it (e.g. saving to 'file.ext' assumes that signac users are using the `with job` to change the current working directory)..

We could reword the comment to "# Add your own code to save barostat_dof to persistent storage." @b-butler what do you think?

Reply:
I've thought about this more. We should either remove the example entirely or include an example of saving/restoring. `numpy` is the lowest common denominator for this. Users of workflows will need to adapt filenames as needed after pasting. I would also use `numpy.save` instead of `savetxt` though to encourage good practices. Agreed?

Reply:
@joaander I think keeping the storage and using NumPy (definitely `save` over `savetxt`) is fine. I also think JSON through `json.dump` is also fine and perhaps another low common denominator.

In general with questions like this, I think showing the saving of the barostat DOF is better than not because it moves users towards restartable jobs over not including it. Examples which have multiple solutions, but emphasize a general best practice are fine to include. Power users will still do what they need and new users wouldn't have even saved the barostat DOF to begin with.

Reply:
It would be ideal to place these generated files in a temp directory (otherwise, they appear in the directory where pytest is executed). Would the examples look OK if they wrote to `path / 'file.extension'`? The definition of `path` would not be shown and up to the user - in pytest I would make it be `tmp_path` https://docs.pytest.org/en/7.3.x/how-to/tmp_path.html#tmp-path.